dd-sdk-android/src/test/kotlin/com/datadog/android/webview/internal/log/WebLogEventDateCorrectorTest.kt

```diff
@@ -0,0 +1,278 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.webview.internal.log
+
+import android.util.Log
+import com.datadog.android.core.internal.time.TimeProvider
+import com.datadog.android.log.internal.logger.LogHandler
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.android.utils.forge.aFormattedTimestamp
+import com.datadog.android.utils.mockSdkLogHandler
+import com.datadog.android.utils.restoreSdkLogHandler
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.argThat
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import java.lang.IllegalArgumentException
+import java.text.Format
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import java.util.concurrent.TimeUnit
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class),
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class WebLogEventDateCorrectorTest {
+    lateinit var testedWebLogEventDateCorrector: WebLogEventDateCorrector
+
+    @Mock
+    lateinit var mockTimeProvider: TimeProvider
+
+    lateinit var originalSdkLogHandler: LogHandler
+
+    @Mock
+    lateinit var mockSdkLogHandler: LogHandler
+
+    @BeforeEach
+    fun `set up`() {
+        testedWebLogEventDateCorrector = WebLogEventDateCorrector(mockTimeProvider)
+        originalSdkLogHandler = mockSdkLogHandler(mockSdkLogHandler)
+    }
+
+    @AfterEach
+    fun `tear down`() {
+        restoreSdkLogHandler(originalSdkLogHandler)
+    }
+
+    @Test
+    fun `M apply the time correction W correctDate`() {
+        // Given
+        val fakeDate = "2021-02-01T20:00:00.000Z"
```

Comment:
Can we try making this test a bit more generic. Here we're only testing we can correct a event recorded on February 2nd 2021, with a 10s offset.